documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -257,27 +265,25 @@ This will be configurable on `Mice` (i.e. via `InputContext.Mice.ClickConfigurat
 Unlike 1.0 and 2.0, this proposal uses `readonly record struct`s as their only argument for the event action. This allows us to provide more information to the event handlers without breaking in the future. These types are farily simple:
 
 ```cs
-public readonly record struct ConnectionEvent(IInputDevice Device, bool IsConnected);
-public readonly record struct KeyDownEvent(IKeyboard Keyboard, Key Key, bool IsRepeat);
-public readonly record struct KeyUpEvent(IKeyboard Keyboard, Key Key);
-public readonly record struct KeyCharEvent(IKeyboard Keyboard, char Character);
-public readonly record struct MouseDownEvent(IMouse Mouse, Vector2 Position, MouseButton Button);
-public readonly record struct MouseUpEvent(IMouse Mouse, Vector2 Position, MouseButton Button);
-public readonly record struct MouseMoveEvent(IMouse Mouse, Vector2 Position, Vector2 Delta);
-public readonly record struct MouseScrollEvent(IMouse Mouse, Vector2 Position, Vector2 WheelPosition, Vector2 Delta);
-public readonly record struct MouseClickEvent(IMouse Mouse, Vector2 Position, MouseButton Button);
-public readonly record struct JoystickDownEvent(IJoystick Joystick, JoystickButton Button);
-public readonly record struct JoystickUpEvent(IJoystick Joystick, JoystickButton Button);
-public readonly record struct JoystickHatMoveEvent(IJoystick, Vector2 Value, Vector2 Delta);
-public readonly record struct JoystickAxisMoveEvent(IJoystick Joystick, int Axis, float Value, float Delta);
-public readonly record struct GamepadDownEvent(IGamepad Gamepad, JoystickButton Button);
-public readonly record struct GamepadUpEvent(IGamepad Gamepad, JoystickButton Button);
-public readonly record struct GamepadThumbstickMoveEvent(IJoystick, Vector2 Value, Vector2 Delta);
-public readonly record struct GamepadTriggerMoveEvent(IJoystick Joystick, int Axis, float Value, float Delta);
+public readonly record struct ConnectionEvent(IInputDevice Device, DateTime Timestamp, bool IsConnected);
```

Comment:
These should be DateTimeOffset and should be nullable and the last parameter of these records. This is to ensure compatibility with backends that do not provide the timestamp at which the event occurred, and also means that this won't need to go through working group approval at all given that it is a modification the structs are explicitly designed for (see the already-approved comment about "This allows us to provide more information to the event handlers without breaking in the future.")

Reply:
we can't provide an timestamp anyway? Maybe it could be wrong for some milliseconds or less, but in the case of don't receive a timestamp, it's not possible to send a DateTime.Now()?

Reply:
> we can't provide an timestamp anyway? Maybe it could be wrong for some milliseconds or less, but in the case of don't receive a timestamp, it's not possible to send a DateTime.Now()?

If the backend doesnt provide timestamps for inputs, i'd rather us not lie about it doing so, if the user needs a timestamp for an event, they can do `Timestamp ?? DateTime.Now()` themselves

Reply:
True, thx